index.d.ts

```diff
@@ -0,0 +1,118 @@
+/**
+ * Which versions of which packages are required.
+ */
+interface Wanted {
+    /**
+     * Required version of Node.js.
+     */
+    node?: string;
+
+    /**
+     * Required version of npm.
+     */
+    npm?: string;
+
+    /**
+     * Required version of yarn.
+     */
+    yarn?: string;
+}
+
+type OnGetVersion = (error: Error | null, info: VersionInfo) => void;
+
+/**
+ * Gets the version of a package.
+ *
+ * @param packageName   Name of the package.
+ * @param onComplete   Handler for the package name.
+ */
+type GetVersion = (packageName: string, onComplete: OnGetVersion) => void;
+
+/**
+ * Extra options to run with.
+ */
+interface Options {
+    /**
+     * @returns The version of a package.
+     */
+    getVersion: GetVersion;
+}
+
+/**
+ * Positive result of checking a program version.
+ */
+interface SatisfiedVersionInfo {
+    /**
+     * Whether the version was known to satisfy its requirements (true).
+     */
+    isSatisfied: true;
+
+    /**
+     * Retrieved version.
+     */
+    version: string;
```

Comment:
Not sure if this was intentional, but you're missing the `wanted` property here (and in `UnsatisfiedVersionInfo` too).

Reply:
Not intentional!

Reply:
Note: `wanted` uses the `semver` typings (https://www.npmjs.com/package/@types/semver). I'll refrain from adding in the complicated `"set"` member from the typings here. In a subsequent PR it should be straightforward to add that as a dev dependency and reference the types.